builder/proxmox/common/step_remove_cloud_init_drive_test.go

```diff
@@ -0,0 +1,106 @@
+package proxmox
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Telmate/proxmox-api-go/proxmox"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+type CloudInitDriveRemoverMock struct {
+	getConfig func() (map[string]interface{}, error)
+	setConfig func(map[string]interface{}) (string, error)
+}
+
+func (m CloudInitDriveRemoverMock) GetVmConfig(*proxmox.VmRef) (map[string]interface{}, error) {
+	return m.getConfig()
+}
+func (m CloudInitDriveRemoverMock) SetVmConfig(vmref *proxmox.VmRef, c map[string]interface{}) (interface{}, error) {
+	return m.setConfig(c)
+}
+
+var _ CloudInitDriveRemover = CloudInitDriveRemoverMock{}
+
+func TestRemoveCloudInitDrive(t *testing.T) {
+	cs := []struct {
+		name                string
+		builderConfig       *Config
+		initialVMConfig     map[string]interface{}
+		getConfigErr        error
+		expectCallSetConfig bool
+		expectedVMConfig    map[string]interface{}
+		setConfigErr        error
+		expectedAction      multistep.StepAction
+	}{
+		{
+			name:          "remove cloud-init drive",
+			builderConfig: &Config{},
+			initialVMConfig: map[string]interface{}{
+				"ide3": "local-zfs:vm-500-cloudinit,media=cdrom",
+			},
+			expectCallSetConfig: true,
+			expectedVMConfig: map[string]interface{}{
+				"ide3": nil,
+			},
+			expectedAction: multistep.ActionContinue,
+		},
+		{
+			name:          "remove cloud-init config options",
+			builderConfig: &Config{},
+			initialVMConfig: map[string]interface{}{
+				"ciuser":       "root",
+				"cipassword":   "$5$FUaXk2yX$2/E0AXhEfGJqxNA6H9ORURyS8WSnsm8uT9S4vvDqwd7",
+				"sshkeys":      "ssh-ed25519%20AAAAC3NzaC1lZDI1NTE5AAAAIEGL6AvL7oe7nQThd8hr6%2FqKeYtQv3oG%2Fur1x2U3AovJ%20packer",
+				"searchdomain": "example.com",
+				"nameserver":   "9.9.9.9 149.112.112.112",
+				"ipconfig0":    "ip=10.0.10.123/24,gw=10.0.10.1",
+			},
+			expectCallSetConfig: true,
+			expectedVMConfig: map[string]interface{}{
+				"ciuser":       nil,
+				"cipassword":   nil,
+				"sshkeys":      nil,
+				"searchdomain": nil,
+				"nameserver":   nil,
+				"ipconfig0":    nil,
+			},
+			expectedAction: multistep.ActionContinue,
+		},
+	}
+
+	for _, c := range cs {
+		t.Run(c.name, func(t *testing.T) {
+			remover := CloudInitDriveRemoverMock{
+				getConfig: func() (map[string]interface{}, error) {
+					return c.initialVMConfig, c.getConfigErr
+				},
+				setConfig: func(cfg map[string]interface{}) (string, error) {
+					if !c.expectCallSetConfig {
+						t.Error("Did not expect SetVmConfig to be called")
+					}
+					for key, val := range c.expectedVMConfig {
```

Comment:
I'm confused by this test, what is it working with? Looking at the code the payload should essentially contain only a `delete` key/val, therefore the other values are nil by default, since no value exists for each key in the `expectedVMConfig`, unless I'm missing something?

Reply:
Oh dear, sorry for the headaches...
I rewrote the tests, don't know what I was thinking, when I wrote them initially.